src/Storefront/Theme/ThemeScripts.php

```diff
@@ -50,12 +53,11 @@ public function getThemeScripts(): array
         $salesChannelId = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_ID);
         $path = $this->themePathBuilder->assemblePath($salesChannelId, $themeId);
 
-        return $this->cache->get('theme_scripts_' . $path, function (ItemInterface $item) use ($themeName) {
-            $themeConfig = $this->pluginRegistry->getConfigurations()->getByTechnicalName($themeName);
+        /** @var SalesChannelContext $salesChannelContext */
+        $salesChannelContext = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_CONTEXT_OBJECT);
```

Comment:
```suggestion:-1+0
        $salesChannelContext = $request->attributes->get(PlatformRequest::ATTRIBUTE_SALES_CHANNEL_CONTEXT_OBJECT);
        if (!$salesChannelContext instanceof SalesChannelContext) {
            return [];
        }
```

var annotations should not be used, as they could lie to PHPStan :sweat_smile: see https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/writing-code-for-static-analysis.md